modules/promotion/src/PromotionStorage.php

```diff
@@ -24,8 +24,27 @@ public function loadValid(OrderTypeInterface $order_type, StoreInterface $store)
     if (empty($result)) {
       return [];
     }
+
+    $usages_query = $this->database->select('commerce_promotion_usage', 'cpu');
```

Comment:
Which reminds me, we should look into the backend_overridable tag for the service.